packages/cli/src/commands/hydrogen/deploy.ts

```diff
@@ -325,7 +328,10 @@ export async function runDeploy(
         envHandle,
       ).branch;
     } else if (envBranch) {
-      userProvidedEnvironmentTag = envBranch;
+      userProvidedEnvironmentTag = findEnvironmentByBranchOrThrow(
+        deploymentData.environments || [],
+        envBranch,
+      ).branch;
```

Comment:
Hey @graygilmore I remember we talked about this before in a previous ticket. We ended up not adding this because we `deploymentData.environments` won't exist if the storefront isn't linked. Do you think it would be okay if we just add it for non-CI?

It's pretty strange that when we do `h2 deploy --env-branch=fake-branch` it just says it works.

Reply:
> Do you think it would be okay if we just add it for non-CI?

Yeah! That makes sense to me. If they're not linked and they run `deploy` are they prompted to link?